src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
-            instance_logger.log("Degree Dist size: " + to_string(degreeDistribution.size()), "info");
-
-            string instanceDataFolderLocation =
-                Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
-            string attributeFilePart = instanceDataFolderLocation + "/" + graphID + "_odd_" + partitionID;
-            ofstream partfile;
-            partfile.open(attributeFilePart, std::fstream::trunc);
-            for (map<long, long>::iterator it = degreeDistribution.begin(); it != degreeDistribution.end(); ++it) {
-                partfile << to_string(it->first) << "\t" << to_string(it->second) << endl;
+                    unordered_set<long> aggregatedEndVidSet = aggregatedCompositeCentralStore[startVid];
+                    aggregatedEndVidSet.insert(endVidSet.begin(), endVidSet.end());
+                    aggregatedCompositeCentralStore[startVid] = aggregatedEndVidSet;
+                }
             }
-            partfile.close();
-        } else if (line.compare(JasmineGraphInstanceProtocol::OUT_DEGREE_DISTRIBUTION) == 0) {
-            instance_logger.log("Received : out degree distribution from server", "info");
+        }
+    }
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string graphID = (data);
-            graphID = Utils::trim_copy(graphID, " \f\n\r\t\v");
-            instance_logger.log("Received Graph ID: " + graphID, "info");
+    for (compositeCentralStoreFileIterator = compositeCentralStoreFileList.begin();
+         compositeCentralStoreFileIterator != compositeCentralStoreFileList.end();
+         ++compositeCentralStoreFileIterator) {
+        std::string compositeCentralStoreFileName = *compositeCentralStoreFileIterator;
+        size_t lastindex = compositeCentralStoreFileName.find_last_of(".");
+        string rawFileName = compositeCentralStoreFileName.substr(0, lastindex);
+        struct stat s;
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+        std::string compositeCentralStoreFile = aggregatorFilePath + "/" + rawFileName;
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partitionID = (data);
-            partitionID = Utils::trim_copy(partitionID, " \f\n\r\t\v");
-            instance_logger.log("Received Partition ID: " + partitionID, "info");
+        if (stat(compositeCentralStoreFile.c_str(), &s) == 0) {
+            if (s.st_mode & S_IFREG) {
```

Comment:
Need to double check why bitwise &

Reply:
It's checking if it's a regular file. It should be bit-wise &.

Reply:
Ok